File: src/main/java/moe/dreameh/assignment1/AddAdvice.java
package moe.dreameh.assignment1;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;

import androidx.appcompat.app.AppCompatActivity;

public class AddAdvice extends AppCompatActivity{
    Bundle bundle = new Bundle();
    private final String[] category = new String[1];

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_advice);

        // Initialization of the spinner
        Spinner spinner = findViewById(R.id.category_spinner);

        // Create an ArrayAdapter using the string array and a default spinner layout
        ArrayAdapter<CharSequence> adapter = ArrayAdapter.createFromResource(this,
                R.array.categories, android.R.layout.simple_spinner_item);

        // Specifiy the layout to use when the list of choices appear
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinner.setAdapter(adapter);



        spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                    category[0] = parent.getItemAtPosition(position).toString();
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }
        });


        // Set cancel result
        final Button buttonCancel = findViewById(R.id.button_cancel);
        buttonCancel.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                setResult(RESULT_CANCELED);
                finish();
            }
        });

        final Button buttonOk = findViewById(R.id.button_create);
        buttonOk.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                EditText authorName = findViewById(R.id.enter_name);
                EditText contentAmount = findViewById(R.id.enter_content);
                if(authorName.getText().length() == 0) {
                    authorName.setError("Field cannot be left blank.");
                } else if(contentAmount.getText().length() == 0) {
                    contentAmount.setError("Field cannot be left blank.");
                } else {
                    addAdviceBundle();

                    Intent resultIntent = new Intent();
                    resultIntent.putExtras(bundle);
                    setResult(RESULT_OK, resultIntent);
                    finish();
                }
            }
        });
    }

    private void addAdviceBundle() {
        // Finding both EditTexts
        EditText authorName = findViewById(R.id.enter_name);
        EditText contentAmount = findViewById(R.id.enter_content);


        bundle.putString("AUTHOR", authorName.getText().toString());
        bundle.putString("CONTENT", contentAmount.getText().toString());
        bundle.putString("CATEGORY", category[0]);
    }


}

File: src/main/java/moe/dreameh/assignment1/AddAdvice.kt
package moe.dreameh.assignment1

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_add_advice.*

class AddAdvice : AppCompatActivity() {
    private var bundle = Bundle()
    private val category = arrayOfNulls<String>(1)

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_advice)

        // Create an ArrayAdapter using the string array and a default spinner layout
        ArrayAdapter.createFromResource(
                this,
                R.array.categories,
                android.R.layout.simple_spinner_item).also { adapter ->
            // Specify the layout to use when the list of choices appears
            adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
            // Apply the adapter to the spinner
            category_spinner.adapter = adapter
        }

        category_spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
                category[0] = parent.getItemAtPosition(position).toString()
            }

            override fun onNothingSelected(parent: AdapterView<*>) {
                // Nothing will be added here.
            }
        }

        // Set cancel result
        button_cancel.setOnClickListener {
            setResult(Activity.RESULT_CANCELED)
            finish()
        }

        button_create.setOnClickListener {
            when {
                enter_name.text.isEmpty() -> enter_name.error = "Field cannot be left blank."
                enter_content.text.isEmpty() -> enter_content.error = "Field cannot be left blank."
                else -> {
                    // Add the contents of the fields into the bundle.
                    addAdviceBundle()

                    setResult(Activity.RESULT_OK, Intent().putExtras(bundle))
                    finish()
                }
            }
        }
    }

    private fun addAdviceBundle() {
        // Insert all the content into the bundle.
        with (bundle) {
            putString("AUTHOR", enter_name.text.toString())
            putString("CONTENT", enter_content.text.toString())
            putString("CATEGORY", category[0])
        }
    }
}
